### PR TITLE
Add doc about lazy pulling and eStargz

### PR DESCRIPTION
**Pull request by ktock**
Should be merged after https://github.com/k3s-io/k3s/pull/2936.

This PR adds doc that talks about the lazy pulling feature of k3s (https://github.com/k3s-io/k3s/pull/2936) and the why/how of using it.

### CODE REVIEW

File: content/k3s/latest/en/advanced/_index.md [hunk 1]
```diff
@@ -394,3 +395,57 @@ If enabled, it is required to disable nm-cloud-setup and reboot the node:
...
+
+## Configure k3s for lazy pulling of eStargz
+
+As shown in the following, `--snapshotter=stargz` option is needed for k3s server and agent.
+
+```
+k3s server --snapshotter=stargz
+```
+
+With this configuration, you can perform lazy pulling for eStagz-formatted images.
```
Comment (dweomer) [inline content/k3s/latest/en/advanced/_index.md:428]:
```suggestion
With this configuration, you can perform lazy pulling for eStargz-formatted images.
```

**Review by dweomer — APPROVED** [review]
LGTM (minus one typo: https://github.com/rancher/docs/pull/3322/files#r709614444)

File: content/k3s/latest/en/advanced/_index.md [hunk 2]
```diff
@@ -394,3 +395,57 @@ If enabled, it is required to disable nm-cloud-setup and reboot the node:
 systemctl disable nm-cloud-setup.service nm-cloud-setup.timer
 reboot
 ```
+
+# Enabling Lazy Pulling of eStargz (Experimental)
+
+## What's lazy pulling and eStargz?
+
+Throughout the container lifecycle, pulling images is known as one of the time-consuming steps in the container lifecycle.
```
Comment (brandond) [inline content/k3s/latest/en/advanced/_index.md:403]:
```suggestion
Pulling images is known as one of the time-consuming steps in the container lifecycle.
```

File: content/k3s/latest/en/advanced/_index.md [hunk 3]
```diff
@@ -394,3 +395,57 @@ If enabled, it is required to disable nm-cloud-setup and reboot the node:
...
+According to [Harter, et al.](https://www.usenix.org/conference/fast16/technical-sessions/presentation/harter),
+
+> pulling packages accounts for 76% of container start time, but only 6.4% of that data is read
+
+For solving this issue, k3s experimentally supports *lazy pulling* as a method of pulling images.
+This allows k3s to start a container without waiting for the completion of pulling the entire image.
+Instead, the necessary chunks of contents (e.g. individual files) are fetched on-demand. 
+Especially for large images, this technique can shorten the container startup latency.
```
Comment (brandond) [inline content/k3s/latest/en/advanced/_index.md:411]:
```suggestion
To address this issue, k3s experimentally supports *lazy pulling* of image contents.
This allows k3s to start a container before the entire image has been pulled.
Instead, the necessary chunks of contents (e.g. individual files) are fetched on-demand. 
Especially for large images, this technique can shorten the container startup latency.
```

File: content/k3s/latest/en/advanced/_index.md [hunk 4]
```diff
@@ -394,3 +395,57 @@ If enabled, it is required to disable nm-cloud-setup and reboot the node:
...
+
+For enabling lazy pulling, the target image needs to be formatted as [*eStargz*](https://github.com/containerd/stargz-snapshotter/blob/v0.6.4/docs/stargz-estargz.md).
```
Comment (brandond) [inline content/k3s/latest/en/advanced/_index.md:413]:
```suggestion
To enable lazy pulling, the target image needs to be formatted as [*eStargz*](https://github.com/containerd/stargz-snapshotter/blob/v0.6.4/docs/stargz-estargz.md).
```

File: content/k3s/latest/en/advanced/_index.md [hunk 5]
```diff
@@ -394,3 +395,57 @@ If enabled, it is required to disable nm-cloud-setup and reboot the node:
...
+This is an OCI-alternative but 100% OCI-compatible image format for lazy pulling.
+Because of the compatibility, eStargz can be pushed to standard container registries (e.g. ghcr.io) as well as this is *still runnable* even on eStargz-agnostic runtimes.
+
+eStargz is developed based on [stargz format proposed by Google CRFS project](https://github.com/google/crfs) but comes with practical features including content verification and performance optimization.
```
Comment (brandond) [inline content/k3s/latest/en/advanced/_index.md:417]:
```suggestion
eStargz is developed based on the [stargz format proposed by Google CRFS project](https://github.com/google/crfs) but comes with practical features including content verification and performance optimization.
```

**Review by brandond — CHANGES_REQUESTED** [review]
Suggested some grammar fixes.

**Comment by ktock** [comment]
@brandond Thank you for the suggestion. Fixed them.

File: content/k3s/latest/en/advanced/_index.md (hunk 1 above)
Comment (ktock) [reply content/k3s/latest/en/advanced/_index.md]: Thank you for the review. Fixed.

**Review by brandond — APPROVED** [review]

**Comment by brandond** [comment]
@catherineluse can you merge?

**Pull request merged**